### ISSUE TITLE
The Core loads assemblies using an API that loads them into the wrong AssemblyLoadContext

### ISSUE THREAD

**Issue by SeanFeldman** [Particular/NServiceBus#5662]
Assemblies are loaded by the `AssemblyScanner` using `Assembly.LoadFrom(assemblyPath)` API. When running within Azure Functions in .NET Core, assemblies scanned and loaded appear to have a different `AssemblyLoadContext`. 

![image](https://user-images.githubusercontent.com/1309622/81978869-5be6ea80-95e9-11ea-9086-02801c346348.png)

vs

![image](https://user-images.githubusercontent.com/1309622/81978951-7b7e1300-95e9-11ea-86c9-9dea06538a9f.png)

That's causing types from the loaded assemblies to fail reference equality check. In turn, that causes message handlers identification to fail and messages not being associated with message handlers. This can be resolved by loading assemblies using a shared `AssemblyLoadContext` when running in .NET Core. 

```c#
#if NETCOREAPP2_1
  var context = AssemblyLoadContext.GetLoadContext(Assembly.GetExecutingAssembly());
  assembly = context.LoadFromAssemblyPath(assemblyPath);
#else
  assembly = Assembly.LoadFrom(assemblyPath);
#endif
```

A spike PR has been created in place: https://github.com/Particular/NServiceBus/pull/5652

**Comment by andreasohlund** [comment]
@Particular/exploration-squad given that this is considered a MUST for v8 I just wanted to check that you have it on your radar? (asking since it doesn't seem to on your squad GH project)

**Comment by SeanFeldman** [comment]
@andreasohlund what do you mean 
> have it on your radar

We've implemented a workaround and will be happy to rip it out when Core implements the right solution.

**Comment by andreasohlund** [comment]
>We've implemented a workaround and will be happy to rip it out when Core implements the right solution.

I thought that ☝️ was something that you wanted to prioritize before v8 ships?

**Comment by SeanFeldman** [comment]
It would be a small task that would require V8 to be in place first.

**Comment by andreasohlund** [comment]
Ah, so no changes would be needed to v8?

**Comment by SeanFeldman** [comment]
A small change of removing the workaround to let core scan assemblies correctly rather than doing that in Functions first. It's a pure workaround to ensure the correct set of assemblies is loaded since doesn't do it right and core's assembly loading is a no-op.

**Comment by bording** [comment]
Fixed by #5913
